util/patch/utils.go

```diff
@@ -60,3 +60,39 @@ func toUnstructured(obj runtime.Object) (*unstructured.Unstructured, error) {
 	}
 	return &unstructured.Unstructured{Object: rawMap}, nil
 }
+
+// unsafeFocusedUnstructured returns a shallow copy of the unstructured object given as input.
+//
+// It's not safe to modify any of the keys in the returned unstructured object, the result should be treated as read-only.
+func unsafeFocusedUnstructured(obj *unstructured.Unstructured, focus patchType, isConditionsSetter bool) *unstructured.Unstructured {
+	// Create the return focused-unstructured object with a preallocated map.
+	res := &unstructured.Unstructured{Object: make(map[string]interface{}, len(obj.Object))}
+
+	// Ranges over the keys of the unstructured object, think of this as the very top level of an object
+	// when submitting a yaml to kubectl or a client.
+	//
+	// These would be keys like `apiVersion`, `kind`, `metadata`, `spec`, `status`, etc.
+	for key := range obj.Object {
+		value := obj.Object[key]
+
+		// Perform a shallow copy only for the keys we're interested in.
+		if key == focus.Key() || preserveUnstructuredKeys[key] {
+			res.Object[key] = value
+		}
+
+		// If we've determined that we're able to interface with conditions.Setter interface,
+		// when dealing with the status patch, we need to remove it from the list of changes,
```

Comment:
nit
```suggestion
		// when dealing with the status patch, we need to remove the conditions from the copied data,
```

Reply:
It's not really a copy, adjusted though a bit